registry/client/auth/authchallenge.go

```diff
@@ -48,8 +48,11 @@ func NewSimpleChallengeManager() ChallengeManager {
 
 type simpleChallengeManager map[string][]Challenge
 
-func (m simpleChallengeManager) GetChallenges(endpoint string) ([]Challenge, error) {
-	challenges := m[endpoint]
+func (m simpleChallengeManager) GetChallenges(endpoint *url.URL) ([]Challenge, error) {
+	endpointCopy := *endpoint
```

Comment:
Guess (&endpoint).String() is not pretty but ehhh, consider it an ignorable nit if you don't agree